source/type/type_interfaces.h

```diff
@@ -283,4 +285,10 @@ struct HasMessages {
     void clean_weak_impacts();
 };
 }  // namespace type
+// trait to access the number of elements in the Mode_e enum
+template <>
+struct enum_size_trait<type::Mode_e> {
```

Comment:
Can you move it close to the enum declaration itself (above, line 100), so we get a chance to update both at the same time?